src/__experimental__/components/decimal/decimal.spec.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import TestUtils from 'react-dom/test-utils';
+import Decimal from './decimal.component';
+import Textbox from '../textbox/textbox.component';
+
+function render(props) {
+  const onChange = props.onChange ? props.onChange : () => true;
+
+  return shallow(
+    <Decimal
+      onChange={ onChange }
+      value={ props.value }
+      precision={ props.precision }
+    />
+  );
+}
+
+function renderIntoDocument(props) {
+  return TestUtils.renderIntoDocument(
```

Comment:
is there a specific reason to use this method?

Reply:
The `formatValue` method checks for the `activeElement` in the `document` property. I was having issues with this test using `shallow` and `mount`, but they were resolved when I started using `renderIntoDocument`, which I took from the specs for the existing `Decimal` component.

Reply:
I can't see a reason it shouldn't work on the enzyme `shallow` render

Reply:
I had an issue with changing `document` when using `shallow`:

```Cannot set property activeElement of [object Document] ```

Which kind of made sense to me. I'm sure there's some hack I could use to get this working, but using `renderIntoDocument` made it trivial, and based on the name, I assumed that had been created with something similar to this problem in mind.

Reply:
Using `mount` actually helped me resolve another issue, so have switched over to that.